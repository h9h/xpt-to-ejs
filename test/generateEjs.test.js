const { generateEjs } = require('../src/generateEjs')

test('just if', () => {
  const ast = [
    {
      "XtendInstruction": {
        "bedingung": "referenceExpr != null-",
      },
    },
  ]

  expect(generateEjs(ast)).toEqual({
    ejs: "<%_ if (referenceExpr != null) { _%>",
    error: [],
    instructions: ['Resolve "referenceExpr != null"']
  })
})

test('just expand', () => {
  const ast = [{
    XtendInstruction: {
      expand: 'Root(environment, "consumer", usingPort.portType, ConnectedProvider(environment))',
    },
  }]

  expect(generateEjs(ast)).toEqual({
    ejs: "<%- include('Root', { args: [environment, \"consumer\", usingPort.portType, ConnectedProvider(environment)] }') %>",
    error: [],
    instructions: ["[EXPAND] Included template 'Root(environment, \"consumer\", usingPort.portType, ConnectedProvider(environment))' - check resolution"]
  })
})

test('generate ejs', () => {
  const ast = [
    {
      "XtendInstruction": {
        "bedingung": "referenceExpr != null-",
      },
    },
    {
      "TemplateText": `
    <assignment header="Reference">`,
    },
    {
      "XtendInstruction": {
        "substitution": "referenceExpr",
      },
    },
    {
      "TemplateText": `</assignment>
      `,
    },
    {
      "XtendInstruction": "ENDIF",
    },
    {
      "TemplateText": `
  </assign>
  <jms jndiConnectionFactory="java:comp/env/jms/DashboardConnectionFactory" bytesMessage="true" timeToLive="`,
    },
    {
      "XtendInstruction": {
        "substitution": "getDashboardMessageTimeToLive()",
      },
    },
    {
      "TemplateText": `">
    <jndiDestination>java:comp/env/jms/DashboardQueue</jndiDestination>
  </jms>
  </fork>
  `,
    },
    {
      "XtendInstruction": "ENDDEFINE",
    }]

  expect(generateEjs(ast)).toMatchSnapshot()
})
