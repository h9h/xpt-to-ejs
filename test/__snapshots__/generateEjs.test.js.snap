// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate ejs 1`] = `
{
  "ejs": "<%if (referenceExpr != nul) {%>
    <assignment header="Reference"><%-referenceExpr%></assignment>
      <% } %>
  </assign>
  <jms jndiConnectionFactory="java:comp/env/jms/DashboardConnectionFactory" bytesMessage="true" timeToLive="<%-getDashboardMessageTimeToLive()%>">
    <jndiDestination>java:comp/env/jms/DashboardQueue</jndiDestination>
  </jms>
  </fork>
  ",
  "error": [
    "not handled: "ENDDEFINE"",
  ],
  "instructions": [
    "Resolve "referenceExpr != nul"",
    "[SUBSTITUTION] Check "referenceExpr"",
    "[SUBSTITUTION] Check "getDashboardMessageTimeToLive()"",
  ],
}
`;
